src/pages/ImportPage.tsx

```diff
@@ -0,0 +1,756 @@
+        setShowNotification("success");
+      })
+      .catch(() => {
+        setIsLoading(false);
+        setShowNotification("error");
+      });
+  };
+
+  function renameKeys(obj: any, newKeys: any) {
+    const keyValues = Object.keys(obj).map((key) => {
+      const newKey = newKeys[key] || key;
+      return { [newKey]: obj[key] };
+    });
+    return Object.assign({}, ...keyValues);
+  }
+
+  return (
+    <ComposedModal
+      size="lg"
+      open
+      onClose={closeModal}
+      preventCloseOnClickOutside
+    >
+      <ModalHeader>
+        <h4>Update Headers</h4>
+      </ModalHeader>
+      <ModalBody>
+        {Object.keys(editData)
+          .filter((key) => key !== "id")
+          .map((key) => (
+            <TextInput
+              key={key}
+              id={key}
+              labelText={key}
+              style={{ marginBottom: "1rem" }}
+              value={editData[key]}
+              onChange={(e) => {
+                setEditData({ ...editData, [key]: e.target.value });
+              }}
+            />
+          ))}
+      </ModalBody>
+      <ModalFooter>
+        <Button kind="secondary" onClick={closeModal}>
+          Cancel
+        </Button>
+        <Button kind="primary" onClick={editHeaders}>
+          Confirm
+        </Button>
+      </ModalFooter>
+    </ComposedModal>
+  );
+};
+
+const DeleteModal = (props: {
+  loadData: () => Promise<void>;
+  closeModal: () => void;
+  setIsLoading: React.Dispatch<React.SetStateAction<boolean>>;
+  targetId: number | number[];
+  setShowNotification: React.Dispatch<
+    React.SetStateAction<"success" | "error" | "">
+  >;
+}) => {
+  const { loadData, closeModal, setIsLoading, targetId, setShowNotification } =
+    props;
+
+  const deleteRow = async () => {
+    closeModal();
+    setIsLoading(true);
+    if (Array.isArray(targetId)) {
+      bulkRemove(targetId)
+        .then(async () => {
+          await loadData();
+          setIsLoading(false);
+          setShowNotification("success");
+        })
+        .catch(() => {
+          setIsLoading(false);
+          setShowNotification("error");
+        });
+    } else {
+      remove(targetId)
+        .then(async () => {
+          await loadData();
+          setIsLoading(false);
+          setShowNotification("success");
+        })
+        .catch(() => {
+          setIsLoading(false);
+          setShowNotification("error");
+        });
+    }
+  };
+
+  return (
+    <ComposedModal
+      size="lg"
+      open
+      onClose={closeModal}
+      preventCloseOnClickOutside
+    >
+      <ModalHeader>
+        <h4>Are you sure you want to delete the selected row(s)?</h4>
+      </ModalHeader>
+      <ModalBody />
+      <ModalFooter>
+        <Button kind="secondary" onClick={closeModal}>
+          Cancel
+        </Button>
+        <Button kind="danger" onClick={deleteRow}>
+          Delete
+        </Button>
+      </ModalFooter>
+    </ComposedModal>
+  );
+};
+
```

Comment:
The Modals share a lot of the logic, What do you think of creating a one single modal component, that can handle the logic of all the modal ? @Yuenmin

Reply:
The change is included in the latest commit.